addon/components/modal/index.js

```diff
@@ -0,0 +1,115 @@
+import Component from '@glimmer/component';
+import { action } from '@ember/object';
+import onTransitionEnd from 'ember-transition-end/utils/on-transition-end';
+import { hasTransitions } from 'ember-modal-service/utils/css-transitions';
+import { inject as service } from '@ember/service';
+import { next } from '@ember/runloop';
+import { buildWaiter } from '@ember/test-waiters';
+
+const openWaiter = buildWaiter('ember-modal-service:open-waiter');
+const closeWaiter = buildWaiter('ember-modal-service:close-waiter');
+
+const isNotDestroyed = (target, key, descriptor) => {
+  const targetMethod = descriptor.value;
+
+  descriptor.value = function (...args) {
+    // istanbul ignore if: lifecycle check.
+    if (this.isDestroyed) {
+      return;
+    }
+
+    return targetMethod.apply(this, args);
+  };
+
+  return descriptor;
+};
+
+export default class ModalComponent extends Component {
+  @service modal;
+
+  get element() {
+    return this.args.element;
+  }
+
+  get model() {
+    return this.args.model;
+  }
+
+  constructor() {
+    super(...arguments);
+
+    next(this, '_open');
+  }
+
+  @isNotDestroyed
+  _safeDidOpen() {
+    this.didOpen && this.didOpen();
```

Comment:
should this be `this.args.didOpen`?

Reply:
No, this function is something like a component life-cycle hook, to notify when the modal is visible.

Was created to extend the modal class and prevent checking if a CSS transition exists and attaching to it using more callbacks:

```js
// components/modal-foo.js
class ModalFoo extends ModalComponent {
  didOpen() {
    // Something to do here...
  }
}

Reply:
@josex2r got it! Thanks for explanation!